### PR TITLE
[core] Enable Processing of CSS into proper JSON

### PR DESCRIPTION
**Pull request by YiseBoge**

### CODE REVIEW

File: snap-css-core/package.json [hunk 1]
```diff
@@ -53,7 +53,9 @@
     "cssbeautify": "^0.3.1",
     "csstree-validator": "^2.0.1",
     "postcss": "^8.3.6",
+    "postcss-css-variables": "^0.18.0",
     "postcss-shorthand-expand": "^1.0.1",
```
Comment (TumsaUmata) [inline snap-css-core/package.json]: Are we still using this?

File: snap-css-core/src/index.ts [hunk 2]
```diff
@@ -15,14 +15,14 @@ class SnapCss {
     (opt, ind) => params.includes(this.keys[ind]) || params.includes('a'))
 
   // eslint-disable-next-line no-console
-  public getCSS(css: string, display: (msg: string[]) => any = console.log, load = Loader.loadCSS) {
-    const result = load(css)
-    display(result[1])
-    return result[0]
+  public getCSS(css: string, display: (msg: string[]) => any | void = console.log, load = Loader.loadCSS) {
+    const [result, message] = load(css)
```
Comment (TumsaUmata) [inline snap-css-core/src/index.ts:19]: I think this is more readable

File: snap-css-core/src/optimize/suggest/json-exporter.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    ['skeleton', ''],
+    // ['tailwind', ''],
+    ['uikit', 'uk-'],
+  ]
+
+  private static getObject(inputPath: string, prefix = '') {
+    const results: object[] = []
+    const [result, message] =  Loader.loadFromFile(inputPath)
+    // eslint-disable-next-line no-console
+    console.log(message)
```
Comment (TumsaUmata) [inline snap-css-core/src/optimize/suggest/json-exporter.ts:26]: Let's remember to comment console messages before committing

File: snap-css-core/src/optimize/suggest/json-exporter.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        results.push(rule.toObject())
+      })
+    })
+    return {prefix: prefix, rules: results}
+  }
+
+  private static parseCSSVars() {
+    JsonExporter.trainingData.forEach(data => {
+      // eslint-disable-next-line no-console
+      console.log('Parsing custom vars for:', data[0])
```
Comment (TumsaUmata) [inline snap-css-core/src/optimize/suggest/json-exporter.ts:37]: Same here

File: snap-css-core/src/css/rule.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ export default class Rule {
   static fromAST(rule: postcss.Rule, padding = '') {
     const result = new Rule(rule.selector, padding)
     rule.walkDecls(decl => {
-      result.declarations.push(Declaration.fromAST(decl))
+      result.declarations = result.declarations.concat(Declaration.fromAST(decl))
```
Comment (TumsaUmata) [inline snap-css-core/src/css/rule.ts]: push() seems to be better in performance than concat(). Take a look

**Review by TumsaUmata — APPROVED** [review]

File: snap-css-core/src/optimize/suggest/json-exporter.ts (hunk 3 above)
Comment (YiseBoge) [reply snap-css-core/src/optimize/suggest/json-exporter.ts:26]: it is cause it requires checking if there is no console in the runtime environment

File: snap-css-core/src/css/rule.ts (hunk 5 above)
Comment (YiseBoge) [reply snap-css-core/src/css/rule.ts]: thanks

**Pull request merged**